src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -126,6 +128,7 @@ impl<Src: OpendalSource> OpendalReader<Src> {
         let reader = op
             .reader_with(&object_name)
             .range(split.offset as u64..)
+            .into_future() // Unlike `rustc`, `try_stream` seems require manual `into_future`.
```

Comment:
What does `rustc` refer to? 👀

Reply:
It refers to the normal `.await` [desugaring](https://doc.rust-lang.org/std/future/trait.IntoFuture.html#await-desugaring) performed by `rustc`, in contrast to how the proc macro `try_stream` transforms the `.await` syntax.

Reply:
This is interesting. Heard about `IntoFuture` a long time ago and finally meet one in practice.

Reply:
`opendal::Operator::reader_with` returns:
* (v0.44.2) `FutureReader` which implements `Future`
* (v0.45.1) `FutureRead` which implements `IntoFuture`